### ISSUE TITLE
fec runtime errors

### ISSUE THREAD

**Issue by Romix-lab** [jgaeddert/liquid-wlan#4]
https://github.com/jgaeddert/liquid-wlan/blob/524a5fcce03580112f45f2a0faff6793c7244827/src/wlan_fec.c#L222
Guess, there is a segfault. 
Function wlan_update_viterbi27_blk reads twice per step enc_bits array, which is 2*8*_dec_msg_len bytes length.
Also there is a segfault on
https://github.com/jgaeddert/liquid-wlan/blob/524a5fcce03580112f45f2a0faff6793c7244827/src/wlan_fec.c#L223
I guess, the real nbits argument is equal 8*_dec_msg_len.

**Comment by jgaeddert** [comment]
Yeah, valgrind is showing a bunch of memory errors. I'll dig into it a bit.
